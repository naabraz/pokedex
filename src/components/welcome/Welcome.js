import React from 'react'

import * as string from './strings'
import {
  WelcomeView,
  WelcomeText
} from './style'

const Welcome = () => (
  <WelcomeView>
    <WelcomeText>
      {string.WelcomeTitle}
    </WelcomeText>
  </WelcomeView>
)

export default Welcome
