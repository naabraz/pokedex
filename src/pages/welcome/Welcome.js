import React from 'react'

import Wrapper from '../../components/Wrapper'
import SpringAnimatedView from '../../components/SpringAnimatedView'
import * as string from './strings'
import {
  WelcomeText,
} from './style'

const Welcome = () => (
  <Wrapper>
    <SpringAnimatedView>
      <WelcomeText>
        {string.WelcomeTitle}
      </WelcomeText>
    </SpringAnimatedView>
  </Wrapper>
)

export default Welcome
