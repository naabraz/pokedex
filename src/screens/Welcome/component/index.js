// @flow

import React from 'react'

import Wrapper from 'components/Wrapper'
import SpringAnimatedView from 'components/SpringAnimatedView'

import * as string from './strings'

import {
  WelcomeText,
  GoToRegionsButton,
  GoToRegionsLabel,
  RegionsIcon,
} from '../style'

import { worldwide } from 'assets/icons'

const IWelcome = {
  navigation: Object,
}

const Welcome = (props: IWelcome) => {
  const { navigation } = props

  return (
    <Wrapper>
      <SpringAnimatedView>
        <WelcomeText>
          {string.WelcomeTitle}
        </WelcomeText>
        <GoToRegionsButton onPress={() => navigation.navigate('Regions')}>
          <GoToRegionsLabel>
            {string.RegionsLabel}
          </GoToRegionsLabel>
          <RegionsIcon source={worldwide} />
        </GoToRegionsButton>
      </SpringAnimatedView>
    </Wrapper>
  )
}

export default Welcome
