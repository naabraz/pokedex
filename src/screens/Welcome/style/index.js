import styled from 'styled-components/native';

import {
  black,
  whiteSmoke,
} from 'themes';

export const WelcomeText = styled.Text`
  font-size: 36
  color: ${whiteSmoke}
`;

export const GoToRegionsButton = styled.TouchableOpacity`
  background-color: ${whiteSmoke}
  margin-top: 50
  align-items: center
  border-radius: 5
  padding: 20px
`;

export const GoToRegionsLabel = styled.Text`
  font-size: 18
  color: ${black}
  margin-bottom: 10px
`;

export const RegionsIcon = styled.Image`
  width: 64
  height: 64
`;
