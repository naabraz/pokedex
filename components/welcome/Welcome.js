import React, {Component} from 'react'

import {WelcomeView, WelcomeText} from './style'

class Welcome extends Component{
  render() {
    return (
      <WelcomeView>
        <WelcomeText>
          Pokedex
        </WelcomeText>
      </WelcomeView>
    )
  }
}

export default Welcome