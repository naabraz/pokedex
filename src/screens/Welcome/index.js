import Welcome from './component'

export default Welcome
