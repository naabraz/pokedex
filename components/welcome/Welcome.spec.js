import React from 'react'
import renderer from 'react-test-renderer'

import { shallow } from 'enzyme'

import Welcome from './Welcome'

describe('Welcome component test', () => {
  it('Should render component correctly', () => {
    const welcome = renderer.create(<Welcome />).toJSON()
    expect(welcome).toMatchSnapshot()
  })

  it('Should shallow', () => {
    const wrapper = shallow(<Welcome />)
  })
})