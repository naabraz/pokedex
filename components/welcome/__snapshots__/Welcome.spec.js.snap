// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome component test Should render component 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#f6bd20",
        "display": "flex",
        "height": "100%",
        "justifyContent": "center",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 36,
        },
      ]
    }
  >
    Pokedex
  </Text>
</View>
`;

exports[`Welcome component test Should render component correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#f6bd20",
        "display": "flex",
        "height": "100%",
        "justifyContent": "center",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 36,
        },
      ]
    }
  >
    Pokedex
  </Text>
</View>
`;
